core/src/core/transaction.rs

```diff
@@ -97,14 +176,28 @@ impl KernelFeatures {
 	/// Write tx kernel features out in v1 protocol format.
 	/// Always include the fee and lock_height, writing 0 value if unused.
 	fn write_v1<W: Writer>(&self, writer: &mut W) -> Result<(), ser::Error> {
-		let (fee, lock_height) = match self {
-			KernelFeatures::Plain { fee } => (*fee, 0),
-			KernelFeatures::Coinbase => (0, 0),
-			KernelFeatures::HeightLocked { fee, lock_height } => (*fee, *lock_height),
-		};
 		writer.write_u8(self.as_u8())?;
-		writer.write_u64(fee)?;
-		writer.write_u64(lock_height)?;
+		match self {
+			KernelFeatures::Plain { fee } => {
+				writer.write_u64(*fee)?;
+				writer.write_empty_bytes(8)?;
+			}
+			KernelFeatures::Coinbase => {
+				writer.write_empty_bytes(16)?;
+			}
+			KernelFeatures::HeightLocked { fee, lock_height } => {
+				writer.write_u64(*fee)?;
+				writer.write_u64(*lock_height)?;
+			}
+			KernelFeatures::NoRecentDuplicate {
+				fee,
+				relative_height,
+			} => {
+				writer.write_u64(*fee)?;
+				writer.write_empty_bytes(6)?;
```

Comment:
is the height written in big endian? if so, should add a comment to that effect.

Reply:
Yes. This is 6 bytes of 00, followed by 2 bytes representing the u16 relative height.
We were originally using a u64 to serialize the u16 relative height which felt too implicit.
Now that we are more explicit with the bytes we should explain why we are doing what we are doing. :+1: